config/manager/usde-config.yaml

```diff
@@ -6,9 +6,54 @@ metadata:
     numaplane.numaproj.io/config: usde-config
 data:
   defaultUpgradeStrategy: "pause-and-drain"
-  pipelineSpecExcludedPaths: |
-    - "lifecycle"
-    - "limits"
-    - "watermark"
-  isbServiceSpecExcludedPaths: |
-    - "jetstream.containerTemplate.resources.limits"
+  pipelineSpecDataLossFields: |
+    - path: spec.edges.from
+    - path: spec.edges.to
+    - path: spec.edges.conditions
+      includeSubfields: true
+    - path: spec.interStepBufferServiceName
+    - path: spec.vertices.source.generator
+    - path: spec.vertices.source.kafka
+    - path: spec.vertices.source.http
+    - path: spec.vertices.source.nats
+    - path: spec.vertices.source.jetstream
+    - path: spec.vertices.source.serving
+    - path: spec.vertices.source.udsource.container.image
+    - path: spec.vertices.source.transformer.container.image
+    - path: spec.vertices.source.transformer.container.command
+    - path: spec.vertices.source.transformer.container.args
+    - path: spec.vertices.source.transformer.container.env
+      includeSubfields: true
+    - path: spec.vertices.source.transformer.container.envfrom
+      includeSubfields: true
+    - path: spec.vertices.source.transformer.builtin
+      includeSubfields: true
+    - path: spec.vertices.sink.log
+    - path: spec.vertices.sink.kafka
+    - path: spec.vertices.sink.blackhole
+    - path: spec.vertices.sink.udsink.container.image
+    - path: spec.vertices.sink.udsink.container.command
+    - path: spec.vertices.sink.udsink.container.args
+    - path: spec.vertices.sink.udsink.container.env
+      includeSubfields: true
+    - path: spec.vertices.sink.udsink.container.envfrom
+      includeSubfields: true
+    - path: spec.vertices.udf.container.image
+    - path: spec.vertices.udf.container.command
+    - path: spec.vertices.udf.container.args
+    - path: spec.vertices.udf.container.env
+      includeSubfields: true
+    - path: spec.vertices.udf.container.envfrom
+      includeSubfields: true
+    - path: spec.vertices.udf.builtin
+      includeSubfields: true
+    - path: spec.sideInputs
+      includeSubfields: true
+    - path: spec.templates
+      includeSubfields: true
+  isbServiceSpecDataLossFields: |
+    - path: spec.jetstream.version
+    - path: spec.redis.native.version
+    - path: spec.redis.external.url
+    - path: spec.redis.external.sentinelUrl
+    - path: spec.redis.external.masterName
```

Comment:
@vigith and @whynowy could you please verify that these lists are correct and sufficient to cover all data loss cases?

Reply:
To clarify, the `includeSubfields` value means "if true, we consider it at risk of data loss if the field or any of its children is being modified; if false, we consider it at risk of data loss if the field is modified but not if just its children are"

Reply:
By the way, @vigith and @whynowy - this list is strictly the "data loss" list. We recognize that there are some changes for which PPND will also not be acceptable so we have an issue to later add a separate list which would indicate the fields for which somebody would either have to do Progressive (i.e. "new isbsvc, new pipeline") or otherwise delete and recreate.